applications/GeoMechanicsApplication/custom_elements/interface_stress_state.cpp

```diff
@@ -0,0 +1,60 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+#include "interface_stress_state.h"
+
+namespace Kratos
+{
+
+Matrix InterfaceStressState::CalculateBMatrix(const Matrix& rDN_DX, const Vector& rN, const Geometry<Node>& rGeometry) const
+{
+    if (rN.empty()) return {};
+
+    Matrix result = ZeroMatrix(GetVoigtSize(), rGeometry.WorkingSpaceDimension() * rGeometry.size());
+
+    for (unsigned int i = 0; i < rGeometry.size() / 2; ++i) {
+        const size_t number_of_u_dofs_per_side = result.size2() / 2;
+
+        result(0, i * rGeometry.WorkingSpaceDimension() + 1)                             = -rN[i];
+        result(0, i * rGeometry.WorkingSpaceDimension() + 1 + number_of_u_dofs_per_side) = rN[i];
+
+        result(1, i * rGeometry.WorkingSpaceDimension())                             = -rN[i];
+        result(1, i * rGeometry.WorkingSpaceDimension() + number_of_u_dofs_per_side) = rN[i];
+    }
+
+    return result;
+}
+
+double InterfaceStressState::CalculateIntegrationCoefficient(const Geometry<Node>::IntegrationPointType& rIntegrationPoint,
+                                                             double                DetJ,
+                                                             const Geometry<Node>& rGeometry) const
+{
+    return rIntegrationPoint.Weight() * DetJ;
+}
+
+Vector InterfaceStressState::CalculateGreenLagrangeStrain(const Matrix& rDeformationGradient) const
+{
+    KRATOS_ERROR << "For interfaces, it is not possible to calculate the Green Lagrange "
+                    "strain based on a deformation gradient.\n";
+}
+
+std::unique_ptr<StressStatePolicy> InterfaceStressState::Clone() const
+{
+    return std::make_unique<InterfaceStressState>();
+}
+
+const Vector& InterfaceStressState::GetVoigtVector() const { return VoigtVectorInterface2D; }
+
+SizeType InterfaceStressState::GetVoigtSize() const { return GetVoigtSizeInterface2D(); }
+
+SizeType InterfaceStressState::GetStressTensorSize() const { return 0; }
```

Comment:
The logical return here might be 1, but needs to be verified

Reply:
The elements of the tensor are yy and xy for normal and shear. Mostly the traction vector will be used, My guess for the stress tensor size would be 2, The fill in of the 2x2 tensor only contains the earlier mentioned elements.